Fix a couple of distribution problems

charris opened the pull request:
- Fix pavement.py to use PEP440 naming for development distributions.
- Add `cblasfuncs.c` and `python_xerbla.c` to `MANIFEST.in`

The last fixes a dependency of source distributions on `HAVE_CBLAS`, which is determined by the local configuration.

charris committed:
BUG: PEP440 compliant versioning for development releases.

The pavement script was not using the new '.dev0+' GIT_REVISION[:7]
suffix for development releases.

charris committed:
BUG: Add cblasfuncs.c and python_xerbla.c to MANIFEST.in

Those files in numpy/core/src/multiarray were only included in source
distributions when HAVE_CBLAS was defined, making the distribution
contents depend on the local configuration.

Closes #6343.

juliantaylor commented on pavement.py:
```diff
@@ -89,7 +89,7 @@
 
     if not setup_py.ISRELEASED:
-        FULLVERSION += '.dev-' + GIT_REVISION[:7]
+        FULLVERSION += '.dev0+' + GIT_REVISION[:7]
```
hm another change to the versioning?
this case is not covered by our NumpyVersion unittests

charris replied:
It's consistent with the `setup.py` file. Might be tricky to test on account of the paver dependency.

charris replied:
PEP440 only requires the `dev0` bit. Wonder what to do about `NumpyVersion`?

charris replied:
Looks like it won't cause a problem
```
In [7]: NumpyVersion('1.9.0.dev-e24486e') == NumpyVersion('1.9.0.dev0+e24486e')
Out[7]: True

In [8]: NumpyVersion('1.9.0.dev-e24486e') == NumpyVersion('1.9.0.dev0')
Out[8]: True
```
However, this in Scipy probably doesn't do what is expected.
```
_rfft_mt_safe = (NumpyVersion(np.__version__) >= '1.9.0.dev-e24486e')
```

charris replied:
`scipy/_lib/tests/test__version.py` looks more up to date.

charris committed:
TST: Update test__version.py.

Added tests for the x.x.dev0+1234567 form  together with alpha and beta
versions of same.

Updates lifted from scipy.

charris commented:
Updated the tests.

charris commented:
OK, putting this in for backport.
